soh/src/code/z_map_exp.c

```diff
@@ -855,60 +858,58 @@ void Minimap_Draw(PlayState* play) {
-                                TopLeftX = OTRGetRectDimensionFromLeftEdge(gMapData->owEntranceIconPosX[sEntranceIconMapIndex]) << 2;
-                                TopLeftY = gMapData->owEntranceIconPosY[sEntranceIconMapIndex] << 2;
-                                TopLeftW = OTRGetRectDimensionFromLeftEdge(gMapData->owEntranceIconPosX[sEntranceIconMapIndex] + IconSize) << 2;
-                                TopLeftH = (gMapData->owEntranceIconPosY[sEntranceIconMapIndex] + IconSize) << 2;
+                                entranceX = OTRGetRectDimensionFromRightEdge(PosX + CVarGetInteger("gMinimapPosX", 0));
+                            } else if (CVarGetInteger("gMinimapPosType", 0) == 3) { // Anchor None
+                                entranceX = PosX + CVarGetInteger("gMinimapPosX", 0);
+                            } else if (CVarGetInteger("gMinimapPosType", 0) == 4) { // Hidden
+                                entranceX = -9999;
                             }
                         }
-                        if (CVarGetInteger("gFixDungeonMinimapIcon", 0) != 0){
-                            //No idea why and how Original value work but this does actually fix them all.
-                            PosY = PosY+1024;
-                        }
 
-                        if (((gMapData->owEntranceFlag[sEntranceIconMapIndex] == 0xFFFF) ||
-                             ((gMapData->owEntranceFlag[sEntranceIconMapIndex] != 0xFFFF) &&
-                              (gSaveContext.infTable[26] & gBitFlags[gMapData->owEntranceFlag[mapIndex]])
-                             )) || CVarGetInteger("gAlwaysShowDungeonMinimapIcon", 0)) {
-                            if (!Map0 || !CVarGetInteger("gFixDungeonMinimapIcon", 0)) {
-                                gDPLoadTextureBlock(OVERLAY_DISP++, gMapDungeonEntranceIconTex, G_IM_FMT_RGBA, G_IM_SIZ_16b,
-                                                    IconSize, IconSize, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP,
-                                                    G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
-                                gSPWideTextureRectangle(OVERLAY_DISP++, TopLeftX, TopLeftY, TopLeftW, TopLeftH, G_TX_RENDERTILE, 0, 0, 1 << 10, 1 << 10);
+                        // For icons that normally would be placed in 0,0 leave them there based on the left edge dimension
+                        // or hide them entirely based on the settings
+                        if (gMapData->owEntranceIconPosY[sEntranceIconMapIndex] == 0) {
+                            entranceY = 0;
+                            if (CVarGetInteger("gFixDungeonMinimapIcon", 0) || CVarGetInteger("gMinimapPosType", 0) == 4) { // Hidden
+                                entranceX = -9999;
+                            } else {
+                                entranceX = OTRGetRectDimensionFromLeftEdge(0);
                             }
```

Comment:
if we have `PosType == 4` we already would have set `entranceX` on L887, so could we just replace this block with

```suggestion
                            entranceX = CVarGetInteger("gFixDungeonMinimapIcon", 0) ? -9999 : OTRGetRectDimensionFromLeftEdge(0);
```

Reply:
That wouldn't work, as if the fix is off, then `entranceX` becomes 0 (relative to left edge) making it no longer hidden. The goal is to hide the icon if the fix is applied, or if posType is hidden.

Unless you meant something more like
```c
if (gMapData->owEntranceIconPosY[sEntranceIconMapIndex] == 0 && entranceX != -9999) {
    entranceY = 0;
    entranceX = CVarGetInteger("gFixDungeonMinimapIcon", 0) ? -9999 : OTRGetRectDimensionFromLeftEdge(0);
}
```

But I'm not sure if that is cleaner/more readable

Reply:
i'm trying to understand the logic behind `entranceX`

on L887 the logic is "no matter what, if `CVarGetInteger("gMinimapPosType", 0) == 4`, set it to `-9999`"

then down here we check to see if it's a 0,0 icon. if it is, we override whatever happened before and set it.

i'll pull the branch and see if i can come up with something that works/makes sense outside of the gh ui real quick, if not this should be good to go

Reply:
> on L887 the logic is "no matter what, if CVarGetInteger("gMinimapPosType", 0) == 4, set it to -9999"

this is true but then later on it is "if the original Y is 0 then override the values". So if we want to preserve the "hidden" PosType, we have to keep check that again somehow.

Honestly the more I think about it, the more it might be cleaner to just move the "hidden" check in the same spot where the "press L to disable minimap" logic is, as that is really what it should be doing